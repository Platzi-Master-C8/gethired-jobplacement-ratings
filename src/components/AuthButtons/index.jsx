import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
