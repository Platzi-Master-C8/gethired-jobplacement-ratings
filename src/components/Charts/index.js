import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }