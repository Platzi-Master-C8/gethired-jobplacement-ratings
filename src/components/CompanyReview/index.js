import CompanyReview from './CompanyReview';

export { CompanyReview };
