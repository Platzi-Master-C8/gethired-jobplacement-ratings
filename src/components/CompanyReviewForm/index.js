import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
