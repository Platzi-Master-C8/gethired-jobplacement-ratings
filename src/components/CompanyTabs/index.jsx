import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
