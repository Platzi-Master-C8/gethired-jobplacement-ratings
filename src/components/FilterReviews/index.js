import FilterReviews from './FilterReviews';

export {
    FilterReviews
}