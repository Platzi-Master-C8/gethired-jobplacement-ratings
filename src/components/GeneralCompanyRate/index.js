import GeneralCompanyRate from './GeneralCompanyRate';

export { GeneralCompanyRate };
