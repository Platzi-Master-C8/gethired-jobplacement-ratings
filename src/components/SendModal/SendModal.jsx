import React from 'react';
import PropTypes from 'prop-types';
import { Box, Button, CircularProgress, Grid, Modal, Typography } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';

const boxStyles = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: '30%',
    height: '20%',
    bgcolor: 'white',
    boxShadow: 24,
    borderRadius: '12px',
    overflow: 'auto',
};

const closeButton = {
    display: 'flex',
    justifyContent: 'flex-end',
};

const SendModal = ({ open, loading, error, handleClose }) => {
    if (loading) {
        return (
            <Modal
                open={open}
                onClose={handleClose}
                aria-labelledby="send-modal-title"
                aria-describedby="send-modal-description"
            >
                <Box sx={boxStyles}>
                    <Grid item sx={closeButton}>
                        <Button variant="text" onClick={handleClose}>
                            <CloseIcon />
                        </Button>
                    </Grid>
                    <Grid
                        container
                        spacing={2}
                        direction="column"
                        alignItems="center"
                        justifyContent="center"
                        sx={{ padding: '20px' }}
                    >
                        <Grid item>
                            <Typography variant="subtitle1">Send applicant review</Typography>
                        </Grid>
                        <Grid item>
                            <CircularProgress color="secondary" />
                        </Grid>
                    </Grid>
                </Box>
            </Modal>
        );
    }

    return (
        <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="send-modal-title"
            aria-describedby="send-modal-description"
        >
            <Box sx={boxStyles}>
                <Grid item sx={closeButton}>
                    <Button variant="text" onClick={handleClose}>
                        <CloseIcon />
                    </Button>
                </Grid>
                {error && (
                    <Grid
                        container
                        spacing={2}
                        direction="column"
                        alignItems="center"
                        justifyContent="center"
                        sx={{ padding: '20px' }}
                    >
                        <Grid container item spacing={1}>
                            <Grid item>
                                <Typography variant="subtitle1">Â¡Error!: Try again</Typography>
                            </Grid>
                            <Grid item>
                                <ErrorOutlineIcon color="error" />
                            </Grid>
                        </Grid>
                        <Grid container item>
                            <Typography variant="subtitle2">There was an error when sending the review</Typography>
                        </Grid>
                    </Grid>
                )}
                {!error && (
                    <Grid
                        container
                        spacing={2}
                        direction="column"
                        alignItems="center"
                        justifyContent="center"
                        sx={{ padding: '20px' }}
                    >
                        <Grid container item spacing={1}>
                            <Grid item>
                                <Typography variant="subtitle1">Submitted successfully</Typography>
                            </Grid>
                            <Grid item>
                                <CheckCircleOutlineIcon color="success" />
                            </Grid>
                        </Grid>
                        <Grid container item>
                            <Typography variant="subtitle2">
                                The review will soon be able in the company page
                            </Typography>
                        </Grid>
                    </Grid>
                )}
            </Box>
        </Modal>
    );
};

SendModal.propTypes = {
    open: PropTypes.bool.isRequired,
    loading: PropTypes.bool.isRequired,
    error: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
};

export default SendModal;
