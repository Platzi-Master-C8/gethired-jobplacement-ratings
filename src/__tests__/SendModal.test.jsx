import React, { useState } from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { SendModal } from '../components/SendModal';

const defaultProps = {
    open: true,
    error: false,
    loading: false,
    handleClose: () => {},
};

const renderComponent = (props = {}) => render(<SendModal {...defaultProps} {...props} />);

describe('<SendModal />', () => {
    test('render Component', () => {
        const { asFragment } = renderComponent({});
        expect(asFragment()).toMatchSnapshot();
    });

    test('Modal loading', () => {
        const { getByText, getByTestId, getByRole } = renderComponent({ loading: true });

        expect(getByTestId(/closeicon/i)).toBeTruthy();
        expect(getByText(/send applicant review/i)).toBeTruthy();
        expect(getByRole('progressbar')).toBeTruthy();
    });

    test('Modal error', () => {
        const { getByText, getByTestId } = renderComponent({ error: true });

        expect(getByTestId(/closeicon/i)).toBeTruthy();
        expect(getByText(/Â¡error!: Try again/i)).toBeTruthy();
        expect(getByText(/there was an error when sending the review/i)).toBeTruthy();
    });

    test('Modal success', () => {
        const { getByText, getByTestId } = renderComponent({});

        expect(getByTestId(/closeicon/i)).toBeTruthy();
        expect(getByText(/Thank you for your time!/i)).toBeTruthy();
        expect(getByText(/Return/i)).toBeTruthy();
    });

    test('Modal close', () => {
        const Component = () => {
            const [open, setOpen] = useState(true);
            const error = false;
            const loading = false;
            return (
                <React.Fragment>
                    <h1>Test</h1>
                    <SendModal open={open} error={error} loading={loading} handleClose={() => setOpen(false)} />
                </React.Fragment>
            );
        };
        const { getByText, getByTestId } = render(<Component />);

        expect(getByTestId(/closeicon/i)).toBeTruthy();
        fireEvent.click(getByTestId(/closeicon/i));
        expect(getByText(/test/i)).toBeTruthy();
    });

    test('Modal close with return button', () => {
        const Component = () => {
            const [open, setOpen] = useState(true);
            const error = false;
            const loading = false;
            return (
                <React.Fragment>
                    <h1>Test</h1>
                    <SendModal open={open} error={error} loading={loading} handleClose={() => setOpen(false)} />
                </React.Fragment>
            );
        };
        const { getByText } = render(<Component />);

        expect(getByText(/Return/i)).toBeTruthy();
        fireEvent.click(getByText(/Return/i));
        expect(getByText(/test/i)).toBeTruthy();
    });
});
