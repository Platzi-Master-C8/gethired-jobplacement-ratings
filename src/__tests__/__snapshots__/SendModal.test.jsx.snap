// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendModal /> render Component 1`] = `<DocumentFragment />`;
